compiler/rustc_session/src/options.rs

```diff
@@ -1463,8 +1477,10 @@ options! {
         "output statistics about monomorphization collection"),
     dump_mono_stats_format: DumpMonoStatsFormat = (DumpMonoStatsFormat::Markdown, parse_dump_mono_stats, [UNTRACKED],
         "the format to use for -Z dump-mono-stats (`markdown` (default) or `json`)"),
-    dump_solver_proof_tree: bool = (false, parse_bool, [UNTRACKED],
-        "dump a proof tree for every goal evaluated by the new trait solver"),
+    dump_solver_proof_tree: SolverProofTreeCondition = (SolverProofTreeCondition::Never, parse_solver_proof_tree_condition, [UNTRACKED],
+        "dump a proof tree for every goal evaluated by the new trait solver. The default is `always`"),
```

Comment:
is the default always? it says `SolverProofTreeCondition::Never` above. also, maybe list them all like "(`always`, `on-error`, or defaults to `never`)" or something along those lines?

Reply:
Uhhh, I think that `-Zdump-solver-proof-tree` defaults to `always` but if you don't write the flag at all it defaults to never. That's why this looks sus lol